### ISSUE TITLE
Panel issue temporary install script for Scales

### ISSUE THREAD

**Issue by Omeryl** [PufferPanel/PufferPanel#545]
I'd love to be able to point my scales machine to my panel with a one time use key to auto provision like the following:

```
To install Scales on your server, run this: 
curl http://mypanel/scales/install/sdfJKLJfdsh98h9812h3jhkjfds | sudo bash
```

where it'd generate a simple shell script with everything it needed to do, including the npm install, installing deps, and what have you.

**Comment by delta360** [comment]
There is a installer I've made on https://github.com/delta360/PufferPanel-Better-Installer which works in a similar way, although I'm not sure on Socket.io connectivity that PufferPanel and Scales use, and I'm not sure you would be able to integrate this kind of provisional system into Scales at the moment, but let the developers answer that.

**Comment by Omeryl** [comment]
Essentially, all this would do is just download scales, npm install, inject the config.js. It's very possible, and this was submitted as a feature request after being asked to by the devs in IRC.

**Comment by Omeryl** [comment]
Modified my original comment to make it easier to understand, :).

**Comment by delta360** [comment]
Yes but what I'm meaning is that Scales relies on Socket.io Connection via HTTPS/SSL which means that you would have to generate self-signed (unless you have a wildcard) certificates which the browsers do not like and manually activate them each time you enter server management since it's the client browser that connects to the server and not the CP itself.

**Comment by Omeryl** [comment]
Yeah, I see what you mean, but it still can prompt for these in the end. Generate self signed by default, have them replaced later. It still does a bunch of the installation ;).

**Comment by LordRalex** [comment]
I dislike how you just simply made a "better installer" and are not even recommending anything upstream. This kinda defeats part of the purpose of open source software, and makes me personally dislike putting stuff on a quite (annoying) liberal license. While I appreciate that it is getting looked at, calling "Better Installer" is quite discouraging, considering a huge base of it is using the original installer.

**Comment by Omeryl** [comment]
Yeah.. you should really consider fixing your license @delta360, that's discouraging that you've messed with it.

**Comment by delta360** [comment]
What do you mean, the license was the one that Github provided, I haven't touched it from the repo creation.

**Comment by Omeryl** [comment]
@delta360 Sure, I suppose, but it wasn't the license for the PufferPanel code (Which is GPL3, you used GPL2). I fixed it, and you merged it, so we're all good :).

**Comment by Omeryl** [comment]
I'll start working on this this weekend, assuming all is well.

**Comment by Omeryl** [comment]
Proposed solution to @delta360's problem:

Ask the user upfront whether to generate a self signed certificate, or if they have something of their own:

* `Do you want to use your own certificate, or generate a self signed certificate? [own/generate] `
* `Do you need us to generate a CSR for you to get your own signed certificate? `

**Comment by SonarBeserk** [comment]
A standalone script would be a much better idea imo. Should open a pr on https://github.com/PufferPanel/Tools instead I think.

**Comment by Omeryl** [comment]
I disagree, it makes much more sense to do it this way so we can automatically inject a config amongst some other things.

**Comment by DaneEveritt** [comment]
Closed in 7f997a2 and subsequent bug fixes in ff09bf3.

Thanks for the suggestion!
